underlay/src/main/java/bio/terra/tanagra/filterbuilder/impl/core/SurveyFilterBuilder.java

```diff
@@ -41,21 +39,21 @@ protected List<String> entityGroupIds() {
   }
 
   @Override
-  protected Map<Literal, String> selectedIdsAndEntityGroups(String serializedSelectionData) {
-    Map<Literal, String> idsAndEntityGroups = new HashMap<>();
+  protected List<SelectionItem> selectedIdsAndGroups(String serializedSelectionData) {
+    List<SelectionItem> idsAndGroups = new ArrayList<>();
     DTSurvey.Survey selectionData = deserializeData(serializedSelectionData);
-    for (DTSurvey.Survey.Selection selectedId : selectionData.getSelectedList()) {
-      if (selectedId.hasKey()) {
-        idsAndEntityGroups.put(
-            Literal.forInt64(selectedId.getKey().getInt64Key()), selectedId.getEntityGroup());
+    if (selectionData != null) {
+      for (DTSurvey.Survey.Selection selectedId : selectionData.getSelectedList()) {
```

Comment:
minor / nit: this block can be simplified.. same in other implementation class as well

if (selectionData == null) {
return List.of(); // empty immutable list
}
return selectionData.getSelectedList().stream().filter(id -> id.hasKey()).map(id -> new selectionItem...).toList();